src/DeferredStream.php

```diff
@@ -15,6 +15,13 @@ class DeferredStream
      */
     public static function create(ParserInterface $parser)
     {
+        if ($parser->isDone()) {
+            return \React\Promise\resolve([
```

Comment:
Actually that will be better, was trying to avoid exceptions a bit overzealous . Will refactor that shortly